src/app/batch_generate_keys/batch_generate_keys.ml

```diff
@@ -159,10 +206,15 @@ let output_there_and_back_cmds =
            "NUM_MILLISECONDS Interval that the rate-limiter is applied over \
             (default: 300000)"
          (optional_with_default 300000 int)
-     and sender_key =
-       flag "--sender-pk" ~aliases:[ "sender-pk" ]
+     and origin_sender_public_key =
+       flag "--origin-sender-pk" ~aliases:[ "origin-sender-pk" ]
          ~doc:"PUBLIC_KEY Public key to send the transactions from"
-         (required string)
+         (optional string)
+     and origin_sender_secret_key =
+       (* TODO: this probably needs to be read in from a file instead of just stuck in plaintext in an argument*)
```

Comment:
The `Secrets` library should allow you to read the privkey from a file, with built-in support for passwords in environment variables etc.